src/components/document/sort-work-view.tsx

```diff
@@ -15,25 +16,40 @@ import "../thumbnail/document-type-collection.scss";
  * Various options for sorting the display are available - by user, by group, by tools used, etc.
  */
 export const SortWorkView: React.FC = observer(function SortWorkView() {
-  const { appConfig, persistentUI, sortedDocuments } = useStores();
-
-  //*************************** Determine Sort Options & State  ***********************************
-  const {tagPrompt} = appConfig;
+  const { appConfig, investigation, persistentUI, problem, sortedDocuments, unit } = useStores();
+  const { tagPrompt } = appConfig;
+  const { docFilter: persistentUIDocFilter } = persistentUI;
   const sortTagPrompt = tagPrompt || ""; //first dropdown choice for comment tags
   const sortOptions = ["Group", "Name", sortTagPrompt, "Bookmarked", "Tools"];
+  const filterOptions: DocFilterType[] = ["Problem", "Investigation", "Unit", "All"];
   const [sortBy, setSortBy] = useState("Group");
+  const [docFilter, setDocFilter] = useState(persistentUIDocFilter);
+
+  const handleDocFilterSelection = (filter: DocFilterType) => {
+    sortedDocuments.setDocFilter(filter);
+    persistentUI.setDocFilter(filter);
+    setDocFilter(filter);
+  };
 
   useEffect(()=>{
+    sortedDocuments.setDocFilter(docFilter);
     if (sortBy === sortTagPrompt){
       sortedDocuments.updateTagDocumentMap();
     }
```

Comment:
This can probably be removed. The tags or strategies should now be pulled from the metadata document so there isn't a need to query the comments and build update the tag document map.

Reply:
Right again. I've removed that and the `updateTagDocumentMap` in `SortedDocuments`. It wasn't being called from anywhere else.